vm_insnhelper.c

```diff
@@ -429,7 +429,7 @@ vm_push_frame(rb_execution_context_t *ec,
 
     ec->cfp = cfp;
 
-    if (VMDEBUG == 2) {
+    if (VMDEBUG == 2 && rb_current_execution_context(false) != NULL) {
```

Comment:
Without this I get:

```
../vm_core.h:1988: Assertion Failed: rb_current_execution_context:!expect_ec || ec != NULL
```

It happens during vm init, pushing a dummy frame before `ruby_current_ec` has been set (Init_BareVM -> th_init -> rb_ec_initialize_vm_stack -> vm_push_frame).